### ISSUE TITLE
Did I lose my funds? Tried to fund channel with RTL from lightning wallet yet it says still pending with txid confirmed on blockchain.

### ISSUE THREAD

**Issue by pscopic** [Ride-The-Lightning/RTL#629]
How did this happen? I've a sinking feel I've lost my funds since after looking up txid on blockchain it shows it went through.

What I did:

I had funds in my lightning wallet that I was trying to get into a channel. I went to RTL, hit the channel button, and up pops the check where I input the channel ID and amount.

The remote node op told me later I shouldn't have had to fill that out, it should have been automatic to prevent sending to a wrong address or channel. His is the only channels I have open so there would have been no alternative channels I could have sent to.

I did see a dropdown menu when I tried to paste the channel id. into the recipient address before I hit send.

What I don't get is why in RTL interface opening channel would send funds funds to a channel I didn't have open while hanging on 'pending' ?

``` 

$ lncli pendingchannels
{
    "total_limbo_balance": "0",
    "pending_open_channels": [
        {
            "channel": {
                "remote_node_pub": "03fda97bb2b0b54c457b9156a5165e0da4804e14268f500fae51ec1f1b59fde530",
                "channel_point": "c2de53fd366ee87e2fa34bc6b37505db6977718d074781d01b3c3d496ac95457:1",
                "capacity": "1000000",
                "local_balance": "981732",
                "remote_balance": "0",
                "local_chan_reserve_sat": "10000",
                "remote_chan_reserve_sat": "10000",
                "initiator": "INITIATOR_LOCAL",
                "commitment_type": "STATIC_REMOTE_KEY"
            },
            "confirmation_height": 0,
            "commit_fee": "18268",
            "commit_weight": "600",
            "fee_per_kw": "25233"
        }
    ],
```
As you can see it says pending channel but this transaction shows succeeded:

listchains output:
```
{
            "tx_hash": "9f1f9661405ef2ed88f6fdad472a8819534d4157a75715d8cad7306d5c1ae926",
            "amount": "-102636",
            "num_confirmations": 236,
            "block_hash": "00000000000000000004089e726f5c0d8ba848a8070f13a83399d7c99db7d29c",
            "block_height": 674532,
            "time_stamp": "1615699325",
            "total_fees": "2636",
            "dest_addresses": [
                "bc1quhu9v0z295gqy59x8cl3sf22rda8vm22r3s3hr7ah4rvg5ukk7pswkxaag",
                "bc1qjpfeu2l2khq6de0ucaq53kwzjugw2yprzmegr4"
            ],
            "raw_tx_hex": "020000000001018917a3739a7b61cd14d3af2acc9912969e3eed33e108957e51202193ce55ef12010000001716001469b64a29958b6deeafeb21d031f034f14b16e13f0000000002a086010000000000220020e5f8563c4a2d100250a63e3f18254a1b7a766d4a1c611b8fddbd46c45396b783554110000000000016001490539e2beab5c1a6e5fcc74148d9c29710e5102302473044022050a214793921678956e7dc73dfc64462ee4ea58d7ea7ab3e725c087b8b69aaec0220247244af211268640b0d2a8c6ed9ac9b60ff1f04cd8030961911fef9bf0d483801210219f7c70a2cdf425b85244d66d75353951384773d09ecbd51b71c9a9e55d9ac8800000000",
            "label": "0:openchannel:shortchanid-741655777404321792"
}, 
```

**Comment by saubyk** [comment]
Hi @pscopic there is no way you can open a channel to a "Channel ID"
When you opening a channel you have to specify the peer you're opening the channel with *not* the channel ID.
If you put a random Channel ID in the form, it will not open a channel to that, because its expects a peer that you're connected with, not a channel ID.

So if the operation went through, it would have opened a channel with your remote peer. There is no other possibility.

From your description above, I am not able to clearly understand what you're trying to do.
- Were you opening a channel to a peer and you somehow specified a channel id, where its said "Peer Alias"?
Or
- You're trying to send funds to a peer you have a pending channel open with?

Also, I am not able to track any transactions with your channel point "c2de53fd366ee87e2fa34bc6b37505db6977718d074781d01b3c3d496ac95457" in the mempool.

Are you still able to see pending open channel in the UI or CLI?

Need more details and clarity on the scenario, to be able to advise further.

**Comment by pscopic** [comment]
Thanks for your quick response saubyk. Yes that would be the first option of the question you posed. I had connected to the node and was trying to fund the channel. I specified the channel id where it said "Peer Alias". 

And that's just it, when I list pending transactions I was unable to look up the channel point on the block chain. But when I listed the wallet transactions and looked up the tx_hash on block explorer I can see the successful transaction. 

I don't know where that address came from if it is not possible to send outside of peer channels I'm connected with.

**Comment by pscopic** [comment]
This is what I see when I go to pending transactions and pull up details of the hung transfer:

![pending_transaction-hung](https://user-images.githubusercontent.com/24592982/111236290-b5c09400-85af-11eb-8273-ec7b2e07bbb9.png)

**Comment by pscopic** [comment]
When I try to duplicate the problem I see how trying to send to a channel ID doesn't take. It says in red "peer not found in the list" so presumably it wouldn't allow sending to just some random peer "741655777404321792" that was the channel id. Immediately I saw the drop down menu to the one peer I was trying to send to and I thought I had used the proper drop down recipient when I hit send. So how the heck would it end up being sent to 35nZcEYbZPrPA8t1zDK7f7sCNG3KCxvnaM  is what I can't figure out.

**Comment by pscopic** [comment]
It looks like what happened is when I pasted the chanID into the recipient field is it created an alias:

``` $ lncli listchains 

{
            "tx_hash": "9f1f9661405ef2ed88f6fdad472a8819534d4157a75715d8cad7306d5c1ae926",
            "amount": "-102636",
            "num_confirmations": 236,
            "block_hash": "00000000000000000004089e726f5c0d8ba848a8070f13a83399d7c99db7d29c",
            "block_height": 674532,
            "time_stamp": "1615699325",
            "total_fees": "2636",
            "dest_addresses": [
                "bc1quhu9v0z295gqy59x8cl3sf22rda8vm22r3s3hr7ah4rvg5ukk7pswkxaag",
                "bc1qjpfeu2l2khq6de0ucaq53kwzjugw2yprzmegr4"
            ],
            "raw_tx_hex": "020000000001018917a3739a7b61cd14d3af2acc9912969e3eed33e108957e51202193ce55ef12010000001716001469b64a29958b6deeafeb21d031f034f14b16e13f0000000002a086010000000000220020e5f8563c4a2d100250a63e3f18254a1b7a766d4a1c611b8fddbd46c45396b783554110000000000016001490539e2beab5c1a6e5fcc74148d9c29710e5102302473044022050a214793921678956e7dc73dfc64462ee4ea58d7ea7ab3e725c087b8b69aaec0220247244af211268640b0d2a8c6ed9ac9b60ff1f04cd8030961911fef9bf0d483801210219f7c70a2cdf425b85244d66d75353951384773d09ecbd51b71c9a9e55d9ac8800000000",
            "label": "0:openchannel:shortchanid-741655777404321792"
},
```
That last channel "label" seems to spell it out. So I'm assuming this should be filed as a bug? 

```
"So if the operation went through, it would have opened a channel with your remote peer. There is no other possibility."
```
From listchains command the transaction went to a mystery address while creating a new label while leaving my pending transaction in a hung state. Hopefully this can be patched so no one can have the unfortunate experience of sending funds off channel when using the channel button to fund address.

**Comment by saubyk** [comment]
> It looks like what happened is when I pasted the chanID into the recipient field is it created an alias:

@pscopic an alias is not created like that. Alias has to be a node, you cannot randomly create an alias and a pubkey and open a channel to it, just by mistyping something into a form.

Also, when you're opening the channel, the funds are not "going" anywhere. The funds are just getting locked into a channel. The funds will only "go" when you make a lightning payment using that channel.

I am still not clear on your current state. Do not try to link channel opening and on-chain transactions, its confusing and I am not able to clearly understand your current state of affairs. Please tell the below two things from RTL interface:
1. Do you see any pending open channels?
2. Do you see any unconfirmed on-chain transaction balance?

**Comment by pscopic** [comment]
Yeah it was a pending channel so funds were not reflected in wallet but it was unconfirmed from never having hit the blockchain.

It took me awhile to figure out the source of my confusion was because I actually had a second channel to the same node i'd opened just moments earlier accidentally with only 100k sats in it that I was confusing with my attempt to open a larger 1 mil sats channel to that node. So I kept thinking the funds had been zapped from that open channel because 900k sats was missing.

At any rate, when I finally saw it was a zombified pending channel I was able to delete it with lncli abandonchannels after recompiling the feature into the daemon. This returned the missing funds immediately to my wallet. 

Thanks for your assistance.
